gammapy/spectrum/models.py

```diff
@@ -574,30 +577,129 @@ class TableModel(SpectralModel):
         Array with the values of the model at energies ``energy``.
     amplitude : float
         Model amplitude that is multiplied to the supplied arrays. Defaults to 1.
+    intery : boolean
+        interpolation can be done linearly or in logarithm
     """
 
-    def __init__(self, energy, values, amplitude=1):
+    def __init__(self, energy, values, amplitude=1, scale_logy=True):
         from scipy.interpolate import interp1d
         self.parameters = Bunch(amplitude=amplitude)
         self.energy = energy
         self.values = values
-
+        self.scale_logy = scale_logy
+        
         loge = np.log10(self.energy.to('eV').value)
         try:
             self.unit = self.values.unit
-            logy = np.log10(self.values.value)
+            if scale_logy is True:
+                y = np.log10(self.values.value)
+            else:
+                y = self.values.value
         except AttributeError:
             self.unit = u.Unit('')
-            logy = np.log10(self.values)
-
-        self.interplogy = interp1d(loge,
-                                   logy,
+            if scale_logy is True:
+                y = np.log10(self.values)
+            else:
+                y = self.values
+        self.interpy = interp1d(loge,
+                                   y,
                                    fill_value=-np.Inf,
                                    bounds_error=False,
                                    kind='cubic')
 
+    @classmethod
+    def from_file(cls, filename, param):
+        """A Table containing aborbed values from a XSPEC model
+        as a function of energy.
+        Todo:
+        Format of the file should be describes and discussed in
+        https://gamma-astro-data-formats.readthedocs.io/en/latest/index.html
+
+        Examples
+        --------
+        Fill table from a model
+        from gammapy.spectrum.models import TableModel
+        filename = '$GAMMAPY_EXTRA/datasets/ebl/ebl_franceschini.fits.gz'
+        table_model = TableModel.from_file(filename=filename, param=0.3)
+        """
+        energy_col = 'ENERG'
+
+        filename = str(make_path(filename))
+
+        # Check if parameter value is in range
+        table_param = fits.open(filename)['PARAMETERS']
```

Comment:
Can you try and use `astropy.table.Table` here?
We use it almost everywhere else instead of `astropy.io.fits.BinTable` directly.
To read a specific HDU:
```python
Table.read(filename, hdu='MY_HDU_NAME')
```